pandas/_libs/lib.pyx

```diff
@@ -1351,6 +1375,66 @@ cdef class Seen:
             self.datetime_ or self.datetimetz_ or self.nat_ or self.timedelta_
             or self.period_ or self.interval_ or self.numeric_ or self.object_
         )
+    cdef void get_curr_val(self, curr_seen_t *ret_val):
```

Comment:
Does the getter here have to assign to ret_val? I think would be more natural to have the return type be `curr_seen_t` instead of `void`

Reply:
I use the pointer so that Cython can know which fused variant to dispatch to. I don't think it's possible/good to have a variable return type.

Reply:
Cython suggests you should return a value:

https://cython.readthedocs.io/en/latest/src/userguide/language_basics.html#error-return-values

Otherwise you will do a `PyErr_Occurred()` check after each invocation, which is expensive compared to a primitive value check

Reply:
Do I just have to add ``except -1`` to the signature and change the return type to int?

I'm a little confused by the Cython docs ( I assume Cython adds ``return -1`` or something after each exception in the generated code when I do this?).

Reply:
Yea I think you have the right idea in mind. I would explicitly return either 0 or the value itself when all is well. I think Cython will automatically return -1 on exception and check in any caller